src/OnboardingSPA/pages/Steps/GetStarted/SiteTypeSetup/PrimarySite/index.js

```diff
@@ -37,130 +37,138 @@ const StepPrimarySetup = () => {
+		}
 
-	/**This condition fills the data in input box if the saved category isn't a subcategory from the content*/
-	if ( selectedPrimaryCategoryInStore && ! inputCategVal ) {
-		const found = categoriesArray.find(
-			( e ) => e.name === selectedPrimaryCategoryInStore
-		);
-		if ( ! found && selectedPrimaryCategoryInStore !== 'primaryCategory' )
-			changeInputCateg( selectedPrimaryCategoryInStore );
-	}
+		setSiteClassData( siteClassificationsData?.body );
+		changePrimaryCategory( currentData?.data?.siteType?.primary ?? '' );
+
+		if ( currentData?.data?.siteType?.labelPri === 'custom' ) {
+			changePrimaryCategory( '' );
+			setInputFieldValue( currentData?.data?.siteType?.primary );
+		}
+	};
 
-	const handleCategoryClick = ( idxOfElm ) => {
-		changeCategory( idxOfElm );
-		changeInputCateg( '' );
+	/**
+	 * Function which saves data in redux when category name is selected via chips
+	 *
+	 * @param {string} primType
+	 */
+	const handleCategoryClick = ( primType ) => {
+		changePrimaryCategory( primType );
+		setInputFieldValue( '' );
 		const currentDataCopy = currentData;
-		currentDataCopy.data.siteType.primary =
-			content.categories[ idxOfElm ]?.name;
+		currentDataCopy.data.siteType.labelPri = '';
+		currentDataCopy.data.siteType.primary = primType;
 		setCurrentOnboardingData( currentDataCopy );
 	};
 
 	/**
 	 * Function which saves data in redux when category name is put-in via input box
 	 *
-	 * @param  input
+	 * @param {string} input
 	 */
 	const categoryInput = ( input ) => {
-		changeCategory( -1 );
-		changeInputCateg( input?.target?.value );
+		changePrimaryCategory( '' );
+		setInputFieldValue( input?.target?.value );
 		const currentDataCopy = currentData;
+		currentDataCopy.data.siteType.labelPri = 'custom';
 		currentDataCopy.data.siteType.primary = input?.target?.value;
 		setCurrentOnboardingData( currentDataCopy );
 	};
 
+	const primarySiteTypeChips = () => {
+		let idx = 0;
+		const primaryChipList = [];
+
+		const types = siteClassData?.types;
+		for ( const typeKey in types ) {
+			primaryChipList.push(
+				<div
+					key={ types[ typeKey ]?.slug }
+					tabIndex={ idx + 1 }
+					role="button"
+					className={ `${
+						types[ typeKey ].slug === primaryCategory
+							? 'chosenPrimaryCategory '
+							: ''
+					}nfd-card-pri-category` }
+					onClick={ () =>
+						handleCategoryClick( types[ typeKey ].slug )
+					}
+					onKeyDown={ () =>
+						handleCategoryClick( types[ typeKey ].slug )
+					}
+				>
+					<div className="nfd-card-pri-category-wrapper">
+						<span
+							className={ `nfd-card-pri-category-wrapper-icon ${
+								types[ typeKey ].slug === primaryCategory
+									? 'nfd-card-pri-category-wrapper-icon-selected '
+									: ''
+							}` }
+							style={ {
+								backgroundImage: `url(${ types[ typeKey ]?.icon })`,
+							} }
+						></span>
+						<span className="categName">
+							{ types[ typeKey ]?.label }
+						</span>
+					</div>
+				</div>
+			);
+			idx++;
```

Comment:
Improvement: We should use `map` on the slug keys here, noticed that you've been using this pattern in quite a few places when I was working on my JS story 😄, check Will's side note to understand what I am talking about